src/awkward/_v2/index.py

```diff
@@ -165,6 +157,21 @@ def __copy__(self):
     def _nbytes_part(self):
         return self.data.nbytes
 
+    def to_backend(self, backend):
+        if backend == ak._v2._util.backend[ak.nplike.Numpy]:
+            if isinstance(self.nplike, ak.nplike.Numpy):
+                return self
+            return self._to_backend(ak.nplike.Numpy.instance())
+        elif backend == ak._v2._util.backend[ak.nplike.Cupy]:
+            if isinstance(self.nplike, ak.nplike.Cupy):
+                return self
+            return self._to_backend(ak.nplike.Cupy.instance())
+        else:
+            raise ValueError("Possible values for backends are `cpu` or `cuda`.")
```

Comment:
This, and in Content, is where some code duplication can be reduced by having the function in _util.py handle all the different nplike cases. You want to isolate that in one place so that when we need to add more nplikes to the list, we don't forget it in one of these other places—only one place to fix when it needs fixing.